dali/operators/image/crop/bbox_crop.cc

```diff
@@ -847,13 +875,14 @@ class RandomBBoxCropImpl : public OpImplBase<CPUBackend> {
     }
   }
 
-  void FilterByCentroid(const Box<ndim, float> &crop,
-                        std::vector<Box<ndim, float>> &bboxes,
-                        std::vector<int> &indices) {
+  template <class UnaryPredicate>
```

Comment:
I wonder if you have an idea how to test it inside test_random_bbox_crop.py ?

Reply:
Generally the same method used by the original algorithm, essentially rewrite the filtering code in python and check for agreement between the two since themain "new-ness" of this algorithm that is to be tested is the new filter lambda (below). So, an alternative to [filter_by_centroid](https://github.com/NVIDIA/DALI/blob/main/dali/test/python/operator_2/test_random_bbox_crop.py#L153) can be written and the test can be parameterised by a function pointer to either `filter_by_centeroid` or `filter_by_area`.

```c++
[&](const Box<ndim, float>& bbox_) {
   const float intersec_ = volume(intersection(rel_crop, bbox_));
   return intersec_ / volume(bbox_) > bbox_prune_threshold_;
});
```